app/javascript/bundles/Coupons/components/CouponsIndexPage.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import Spinner from 'react-bootstrap/Spinner'
+
+import CouponsList from './CouponsList';
+
+const CouponsIndexPage = (props) => {
+  const [coupons, setCoupons] = useState([]);
+  const [availability, setAvailability] = useState(true);
+  const [loaded, setLoaded] = useState(false);
+
+  const available = () => setAvailability(true);
+  const used = () => setAvailability(false);
+
+  useEffect(() => {
+    getCouponsFromApi();
+    setLoaded(false);
+  }, [availability]);
+
+  const getCouponsFromApi = async () => {
+    const params = {scope: availability};
+
+    await axios
+        .get(props.path.concat('.json'), {params: params},
+        )
+        .then(function(response) {
+          if (response.status === 200) {
```

Comment:
you don't need this because if you wont receive status 200 it will goes to catch block as I remember